### ISSUE TITLE
Support private key as authentication vs file?

### ISSUE THREAD

**Issue by cbetta** [Vonage/vonage-node-sdk#79]
I currently find the library to be a bit inconsistent when it comes to private keys. As a user I am left to write a key to file:

``` js
var Nexmo = require('nexmo');

var nexmo = new Nexmo({
    apiKey: 'key',
    apiSecret: 'secret',
});


nexmo.app.create(
  'First Voice App',
  'voice',
  'http://example.com/answer',
  'http://example.com/event',
  {},
  function(err, res) {
    var appId = res.id;
    var privateKey = res.keys.private_key;
    var privateKeyFile = __dirname + '/' + appId;
    require('fs').writeFileSync(privateKeyFile, privateKey);
  }
);
```

But then when I try to configure the library with a key it requires a filename and reads the file:

``` js
  nexmo = new Nexmo({
      apiKey: 'key',
      apiSecret: 'secret',
      applicationId: appId,
      privateKey: privateKeyFile
    });
```

I somehow think it would make a lot more sense for the initializer to just accept the key itself and not bother itself with reading from disc. This would also make it possible to create an app and then reinitialize the library in a few lines of code without having to write the key to disc.

Final note: I might want to save my API key somewhere else on the disk (another volume, database, etc) so having this option would be useful regardless.

Thoughts @leggetter @tjlytle ?

**Comment by tjlytle** [comment]
@cbetta I missed specifying this in the client spec (will add) but we ran into a similar conversation around the python library. I'll be updating the spec to read that the client libraries **must** accept the key as a string, and **may** provide a way to pass a filename. Reasoning is the former is possible regardless of how you store the key, the latter dictates storage that may be difficult or impossible depending on how you deploy your application.

**Comment by leggetter** [comment]
Node supports Buffer or a string path at present.

**Comment by cbetta** [comment]
@leggetter yet it does not support this:

``` js
var Nexmo = require('nexmo');

var nexmo = new Nexmo({
    apiKey: 'key',
    apiSecret: 'secret',
});


nexmo.app.create(
  'First Voice App',
  'voice',
  'http://example.com/answer',
  'http://example.com/event',
  {},
  function(err, res) {
    var appId = res.id;
    var privateKey = res.keys.private_key;

    nexmo = new Nexmo({
      apiKey: 'key',
      apiSecret: 'secret',
      applicationId: appId,
      privateKey: privateKey
    });
  }
);
```

What am I missing?

**Comment by leggetter** [comment]
It doesn't support the key being passed as a string. Just:
1. String - path to file
2. Buffer - contents of key loaded into a buffer

The suggestion elsewhere has been to check the start of the string to see if it's the key. See https://github.com/Nexmo/nexmo-python/issues/12

**Comment by cbetta** [comment]
@leggetter check, I just figured that out. I was going to suggest similar behaviour for the node library indeed.

**Comment by leggetter** [comment]
Closed via #80
